mypy/meet.py

```diff
@@ -94,12 +94,19 @@ class C(A, B): ...
     if isinstance(s, UnionType):
         return any(is_overlapping_types(t, item)
                    for item in s.items)
+    if isinstance(t, TypeType) and isinstance(s, TypeType):
+        # If both types are TypeType, compare their inner types.
+        return is_overlapping_types(t.item, s.item, use_promotions)
+    elif isinstance(t, TypeType) or isinstance(s, TypeType):
```

Comment:
What about `Type[X]` vs. `object` or `Type[X]` vs. `type` -- they seem to be overlapping?